### PR TITLE
Fix enum backing type always 'string'

### PR DESCRIPTION
**Pull request by DidierLmn**
Every enum will be described as type "string" because the current logic to determine the backing type does not work
```
is_subclass_of($enumClass, \StringBackedEnum::class));
is_subclass_of($enumClass, \IntBackedEnum::class));
```
These are always false, regardless of backing type as you can see at https://3v4l.org/fButY

Moreover the type would be set as "int", which is not valid in OpenApi-Spec.

This MR ensures proper backing detection of enums and sets int-backed enums to type "integer".

### CODE REVIEW

**Review by palgalik — APPROVED** [review]

File: ModelDescriber/EnumModelDescriber.php [hunk 1]
```diff
@@ -17,6 +17,12 @@ public function describe(Model $model, Schema $schema)
             $enums[] = $enumCase->value;
         }
 
+        $reflectionEnum = new \ReflectionEnum($enumClass);
+        if ($reflectionEnum->isBacked() && 'int' === $reflectionEnum->getBackingType()->getName()) {
+            $schema->type = 'integer';
+        } else {
+            $schema->type = 'string';
+        }
         $schema->type = is_subclass_of($enumClass, \IntBackedEnum::class) ? 'int' : 'string';
```
Comment (DjordyKoert) [inline ModelDescriber/EnumModelDescriber.php:26]: I'm not really a fan of the if-else combined with the multi places where `$schema->type` gets set.

```suggestion
        $schema->type = is_subclass_of($enumClass, \IntBackedEnum::class) || ($reflectionEnum->isBacked() && 'int' === $reflectionEnum->getBackingType()->getName())
            ? 'integer' 
            : 'string';
```

**Comment by DjordyKoert** [comment]
Also please rebase on master for CI

File: ModelDescriber/EnumModelDescriber.php (hunk 1 above)
Comment (Book56) [reply ModelDescriber/EnumModelDescriber.php:26]: Adil Ishaq

File: Tests/Functional/Entity/ArticleType81NotBacked.php
Comment (Book56) [inline Tests/Functional/Entity/ArticleType81NotBacked.php:1]: Adil Ishaq

**Comment by DjordyKoert** [comment]
Thank you @DidierLmn!

**Comment by DidierLmn** [comment]
I kinda lost track of this one. Thanks a lot for finishing it off and merging @DjordyKoert!!  :pray:

**Pull request merged**